docs/contributor_guide.rst

```diff
@@ -1,66 +1,110 @@
 Contribution guidelines
-===============================================
+=======================
 
 First of all, thank you for contributing! We welcome contributions of
 all forms, including but not limited to
 
 - Bug fixes
 - New features
 - Documentation
-- More user-friendly syntax error messages
+- Improved error messages that are more user-friendly
 - New example programs
 - Compiler performance patches
 - Minor typo fixes in the documentation, code, comments (please directly make a pull request for minor issues like these)
 
 How to contribute bug fixes and new features
---------------------------------------------------
+--------------------------------------------
 
 Issues marked with `"welcome contribution" <https://github.com/taichi-dev/taichi/issues?q=is%3Aopen+is%3Aissue+label%3A%22welcome+contribution%22>`_ are easy ones to start with.
 
-- Please first leave a note (e.g. *I know how to fix this and would like to help!*) on the issue, so that people know some one is already working on it. ("Prevent redundant work";)
+- Please first leave a note (e.g. *I know how to fix this and would like to help!*) on the issue, so that people know some one is already working on it. This helps prevent redundant work;
 
-- If no lead developer has commented and described a potential solution on the issue, please also briefly describe your plan and wait for a lead developer to reply before you start. ("Keep solutions simple".)
+- If no core developer has commented and described a potential solution on the issue, please briefly describe your plan, and wait for a core developer to reply before you start.
+  This helps keep implementations simple and effective.
 
 High-level guidelines
 ---------------------
 
-- Almost every design decision has pros and cons. Good decisions are good because their pros outweigh their cons. Always think of both sides of your decision.
-- No overkills: always use the *easiest* solutions to solve easy problems, so that you have time and energy for real hard ones.
-- Debugging is hard. Changesets should be small so that sources of bugs can be easily pinpointed.
+- Be pragmatic: practically solving problems is our ultimate goal.
+- No overkills: always use *easy* solutions to solve easy problems, so that you have time and energy for real hard ones.
+- Almost every design decision has pros and cons. A decision is `good` if its pros outweigh its cons. Always think about both sides.
+- Debugging is hard. Changesets should be small so that sourees of bugs can be easily pinpointed.
```

Comment:
Plus what I learnt from GL, keep interacting with reviewers for each small stage, and make multiple PRs for each big stage.

Reply:
Good idea. Added.